int32 x_increase
int32 y_increase 


bool js_button
bool up_button
bool down_button
